C++/C++.sublime-syntax

```diff
@@ -625,6 +632,9 @@ contexts:
         - match: \b(struct)\s+({{identifier}})\b
           captures:
             1: storage.type.c++
+        - match: \benum\b
+          scope: storage.type.c++
+          set: data-structures-enum-definition
```

Comment:
I think this may require some more special casing. With typedef'ed struct, we don't scope the identifier as `entity.name.struct` since the purpose of the typedef is to expose the struct as the type's identifier. I think we'd probably want to do the same in this case, in which case we'd need to consume the identifier after `enum` and move right to a context where we can match the type's name, a comment, or `{`. Tests will definitely help to ensure we are handling the variations of this structure properly.